samples/java/hibernate/src/main/java/com/google/cloud/postgres/HibernateSampleTest.java

```diff
@@ -0,0 +1,260 @@
+package com.google.cloud.postgres;
+
+import com.google.cloud.postgres.models.Albums;
+import com.google.cloud.postgres.models.Concerts;
+import com.google.cloud.postgres.models.HibernateConfiguration;
+import com.google.cloud.postgres.models.Singers;
+import com.google.cloud.postgres.models.Tracks;
+import com.google.cloud.postgres.models.Venues;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import org.hibernate.Session;
+import org.hibernate.query.Query;
+
+public class HibernateSampleTest {
+
+  private static final Logger logger = Logger.getLogger(HibernateSampleTest.class.getName());
+
+  private HibernateConfiguration hibernateConfiguration;
+
+  private List<String> singersId = new ArrayList<>();
+  private List<String> albumsId = new ArrayList<>();
+  private List<Long> tracksId = new ArrayList<>();
+  private List<String> venuesId = new ArrayList<>();
+  private List<String> concertsId = new ArrayList<>();
+
+  public HibernateSampleTest(
+      HibernateConfiguration hibernateConfiguration) {
+    this.hibernateConfiguration = hibernateConfiguration;
+  }
+
+  public void testJPACriteria() {
+    Session s = hibernateConfiguration.openSession();
+    CriteriaBuilder cb = s.getCriteriaBuilder();
+    CriteriaQuery<Singers> singersCriteriaQuery = cb.createQuery(Singers.class);
+    Root<Singers> singersRoot = singersCriteriaQuery.from(Singers.class);
+    singersCriteriaQuery
+        .select(singersRoot)
+        .where(
+            cb.and(cb.equal(singersRoot.get("firstName"), "David"),
+            cb.equal(singersRoot.get("lastName"), "Lee")));
+
+    Query<Singers> singersQuery = s.createQuery(singersCriteriaQuery);
+    List<Singers> singers = singersQuery.getResultList();
+
+    logger.log(Level.INFO, "Listed singer: {0}", singers.size());
+    s.close();
+  }
+
```

Comment:
Could we also add a test for `CriteriaUpdate` and `CriteriaDelete`?

Reply:
Done